Src/AutoFixture.NUnit3/FixedNameTestMethodBuilder.cs

```diff
@@ -0,0 +1,84 @@
+﻿
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using NUnit.Framework.Interfaces;
+using NUnit.Framework.Internal;
+using System.Globalization;
+
+namespace Ploeh.AutoFixture.NUnit3
+{
+    /// <summary>
+    /// Builder that generates <see cref="TestMethod"/> with fixed names.
+    /// This is needed for some test runners such as the Nunit test adaptor for Visual Studio.
+    /// </summary>
+    /// <seealso cref="MutableNameTestMethodBuilder" />
+    public class FixedNameTestMethodBuilder : MutableNameTestMethodBuilder
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="FixedNameTestMethodBuilder"/> class.
+        /// </summary>
+        public FixedNameTestMethodBuilder() { }
+
+        /// <inheritdoc />
+        protected override TestCaseParameters GetParametersForMethod(IMethodInfo method, object[] args, int autoDataStartIndex)
+        {
+            if (method == null)
+            {
+                throw new ArgumentNullException(nameof(method));
+            }
+
+            if (args == null)
+            {
+                throw new ArgumentNullException(nameof(args));
+            }
+
+            var result = base.GetParametersForMethod(method, args, autoDataStartIndex);
+            var methodParameters = method.GetParameters();
+
+            EnsureOriginalArgumentsArrayIsNotShared(result);
+
+            for (int i = autoDataStartIndex; i < result.OriginalArguments.Length; i++)
+            {
+                result.OriginalArguments[i] = new TypeNameRenderer(methodParameters[i].ParameterType);
+            }
+
+            return result;
+        }
+
+        /// <summary>
+        /// In NUnit 3.5+ the Arguments and OriginalArguments properties are containing two different instances,
+        /// not in earlier versions.<BR/>
+        /// Unfortunately Arguments has a private setter and there is not way to change the instance other
+        /// than by using reflection...<BR/>
+        /// When running in NUnit3.5+ the method is supposed to do nothing.
+        /// </summary>
+        private static void EnsureOriginalArgumentsArrayIsNotShared(TestCaseParameters parameters)
+        {
+            if (ReferenceEquals(parameters.Arguments, parameters.OriginalArguments))
+            {
+                var clonedArguments = new object[parameters.Arguments.Length];
```

Comment:
Please make a copy of the `OriginalArguments` array and update that property. It will be more correct, as we update the `OriginalArguments` property, not the `Arguments` one.